packages/react-components/lib/reports/render-selected-component-from-dict.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+interface RenderSelectedComponentProps {
+  selectedKey: string;
+  obj: Record<string, React.ReactElement | null>;
+}
+
+export const RenderSelectedComponentFromDict = (
```

Comment:
afaik, it conditionally renders an item, or nothing if the item doesn't exist. Would it be simpler if we just use
```tsx
{ Object.prototype.hasOwnProperty.call(obj, selectedKey) && obj[key] ? obj[key] : null }
```

Reply:
Yeah I removed this component and added that logic to ReportDashboard